### PR TITLE
Basic backend design

### PR DESCRIPTION
**Pull request by royeom**
Basic design of backend with Flask, Redis and REST API.

New functions are below:

**Register** : Sign-up with informations about username(e-mail), password, first name and last name. And hashed password enters db.
**Login** : Check match of request's password and db's hashed password

Don't mind about `logout.py` and `template` folder; just researching.

Please take a look.

### CODE REVIEW

File: research/backend/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (harrydrippin) [inline research/backend/.vscode/settings.json:1]: `.vscode` is your own preference file for Visual Studio Code. You need to add `.vscode` to `.gitignore`. This code may influence other developer's editor preferences.

File: research/backend/manager/dbmanager.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import redis
```
Comment (harrydrippin) [inline research/backend/manager/dbmanager.py:1]: Why don't you make this code to class representation? It'll be nice for managing and repairing source code.

File: research/backend/manager/dbmanager.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+HOST = "localhost"
+DB = 0
+
+REDIS_DB = redis.StrictRedis(host=HOST, db=DB, charset="utf-8")
```
Comment (harrydrippin) [inline research/backend/manager/dbmanager.py]: Add `decode_responses=True` to your `StrictRedis` declaration parameters. It'll parse your Redis response to UTF-8 format.
Later, I'll provide a framework helps your code to easily interact with Redis.

File: research/backend/restful/login.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+        # Load password from db
+        password_in_db = dbmanager.REDIS_DB.hget('username:' + str(input_username), 'password').decode("utf-8")
+
+        # Check from input and db
+        if bcrypt.checkpw(input_password, password_in_db.encode("utf-8")):
+            return {
+                "result": 0
+            }
+
```
Comment (harrydrippin) [inline research/backend/restful/login.py]: No line between if and else phrase is suggested.

File: research/backend/restful/login.py [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        else:
+            invalid_response = jsonify({
+                "result": 1,
+                "cause": "Id or password is invalid!"
+             })
+            invalid_response.status_code = 401
+            return invalid_response
+
+
```
Comment (harrydrippin) [inline research/backend/restful/login.py]: Just add only **one line** at the end of the code.

File: research/backend/restful/register.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+        # Complete Message
+        return {
+            'message': 'Sign-up is completed.',
+            'username': username,
+            'password': str(password),
+            'firstname': first_name,
+            'lastname': last_name,
+            'hashed': str(hashed)
+            }
```
Comment (harrydrippin) [inline research/backend/restful/register.py]: Add one line after this line.

**Review by harrydrippin — COMMENTED** [review]
Thanks!

You've added lots of binary files (in `__pycache__`) in this commit. You should add `.gitignore` file for removing cache files and not useful codes.

File: research/backend/restful/register.py (hunk 6 above)
Comment (royeom) [reply research/backend/restful/register.py]: Okay, I modified.

File: research/backend/restful/login.py (hunk 5 above)
Comment (royeom) [reply research/backend/restful/login.py]: Okay, I modified

File: research/backend/restful/login.py (hunk 4 above)
Comment (royeom) [reply research/backend/restful/login.py]: Okay, I modified

File: research/backend/manager/dbmanager.py (hunk 3 above)
Comment (royeom) [reply research/backend/manager/dbmanager.py]: Okay, I added a parameter.

Comment (royeom) [reply research/backend/manager/dbmanager.py]: Thanks for framework!!

File: research/backend/.vscode/settings.json (hunk 1 above)
Comment (royeom) [reply research/backend/.vscode/settings.json:1]: Okay, I fixed it!

File: research/backend/manager/dbmanager.py (hunk 2 above)
Comment (royeom) [reply research/backend/manager/dbmanager.py:1]: I expect you to receive the framework  :)
I'll apply it to mine !!

File: research/backend/restful/login.py [hunk 7]
```diff
@@ -29,13 +29,11 @@ def post(self):
...
             invalid_response = jsonify({
                 "result": 1,
                 "cause": "Id or password is invalid!"
              })
             invalid_response.status_code = 401
-            return invalid_response
 
-    
\ No newline at end of file
+            return invalid_response
```
Comment (harrydrippin) [inline research/backend/restful/login.py]: Leave one line at the end of the file next time. This rule is common for all codes here.

File: research/backend/restful/register.py [hunk 8]
```diff
@@ -44,4 +44,4 @@ def post(self):
             'firstname': first_name,
             'lastname': last_name,
             'hashed': str(hashed)
-            }
\ No newline at end of file
+            }
```
Comment (harrydrippin) [inline research/backend/restful/register.py]: Same as above. ;)

**Review by harrydrippin — APPROVED** [review]
Good! Take a look at my additional comments and apply to next PR! 😄

**Pull request merged**